client/src/AppBundle/Model/Sirius/QueuedChecklistData.php

```diff
@@ -24,4 +44,130 @@ public function setChecklistId(int $checklistId): QueuedChecklistData
         return $this;
     }
+
+    /**
+     * @return string|null
+     */
+    public function getChecklistUuid(): ?string
+    {
+        return $this->checklistUuid;
+    }
+
+    /**
+     * @param string|null $checklistUuid
+     * @return QueuedChecklistData
+     */
+    public function setChecklistUuid(?string $checklistUuid): QueuedChecklistData
+    {
+        $this->checklistUuid = $checklistUuid;
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCaseNumber(): string
+    {
+        return $this->caseNumber;
+    }
+
+    /**
+     * @param string $caseNumber
+     * @return QueuedChecklistData
+     */
+    public function setCaseNumber(string $caseNumber): QueuedChecklistData
+    {
+        $this->caseNumber = $caseNumber;
+        return $this;
+    }
+
+    /**
+     * @return \DateTime
+     */
+    public function getReportStartDate(): \DateTime
+    {
+        return $this->reportStartDate;
+    }
+
+    /**
+     * @param \DateTime $reportStartDate
+     * @return QueuedChecklistData
+     */
+    public function setReportStartDate(\DateTime $reportStartDate): QueuedChecklistData
+    {
+        $this->reportStartDate = $reportStartDate;
+        return $this;
+    }
+
+    /**
+     * @return \DateTime
+     */
+    public function getReportEndDate(): \DateTime
+    {
+        return $this->reportEndDate;
+    }
+
+    /**
+     * @param \DateTime $reportEndDate
+     * @return QueuedChecklistData
+     */
+    public function setReportEndDate(\DateTime $reportEndDate): QueuedChecklistData
+    {
+        $this->reportEndDate = $reportEndDate;
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getChecklistFileContents(): string
+    {
+        return $this->checklistFileContents;
+    }
+
+    /**
+     * @param string $checklistFileContents
+     * @return QueuedChecklistData
+     */
+    public function setChecklistFileContents(string $checklistFileContents): QueuedChecklistData
+    {
+        $this->checklistFileContents = $checklistFileContents;
+        return $this;
+    }
+
+    /**
+     * @return array|null
+     */
+    public function getReportSubmissions(): ?array
+    {
+        return $this->reportSubmissions;
+    }
+
+    /**
+     * @param array|null $reportSubmissions
+     * @return QueuedChecklistData
+     */
+    public function setReportSubmissions(?array $reportSubmissions): QueuedChecklistData
+    {
+        $this->reportSubmissions = $reportSubmissions;
+        return $this;
+    }
+
+    /**
+     * @return ReportSubmission|null
+     */
+    public function getSyncedReportSubmission(): ?ReportSubmission
+    {
+        if (null === $this->getReportSubmissions()) {
+            return null;
+        }
+
+        foreach ($this->getReportSubmissions() as $submission) {
+            if ($submission->getUuid()) {
+                return $submission;
+            }
+        }
+
+        return null;
+    }
```

Comment:
We may have an issue with this approach - its whats been leading to so many delays with fixing the DocumentSync bug to do with folders forming incorrectly in Sirius. I'd originally used this pattern but when you start to throw in re-submissions you can't tell which submission with a UUID you should be using.

I *think* I've come up with a solution in DDPB-3438 and I don't think we should delay merging this but we just need to be aware that we'll need to come back and put a fix in place once we've there's a working solution.

Reply:
Yes, I came across that realisation too. I think we are okay here though as the submission UUID in the checklist endpoint isn't actually used for anything - Sirius don't map checklists to reports on their side, and in the case of paper checklists we are using a dummy all 9's value. So I thought it would suffice to use any valid one that we found, else default to all 9's if this returns null.

Reply:
Ah nice - thats a relief!